typegraph/core/wit/typegraph.wit

```diff
@@ -15,12 +15,14 @@ interface core {
     finalize-typegraph: func() -> result<string, error>
 
     type type-id = u32
-
     record type-base {
         name: option<string>,
+        // json
+        injection: option<string>,
```

Comment:
can we type with an enum the different injection?

Reply:
wit enums cannot have type payloads like in rust.

Reply:
But WIT have `variant`, see [WIT types](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md#wit-types).